### PR TITLE
Parse azimuth fm rate polynomial

### PR DESCRIPTION
**Pull request by seongsujeong**
This PR is in regards to #51 when loading azimuth FM rate polynomial from LADS .xml files. It was found that some S1 SAFE data with old IPF has different format for the polynomial. The examples are as below:

S1A_IW_SLC__1SSV_20150114T015005_20150114T015035_004161_0050BB_87F4 (IPF `2.36`)

```xml
<azimuthFmRate>
    <azimuthTime>2015-01-14T01:50:05.584900</azimuthTime>
    <t0>5.339831264815916e-03</t0>
    <c0>-2.327628769423934e+03</c0>
    <c1>4.493262758608217e+05</c1>
    <c2>-7.852728113753408e+07</c2>
</azimuthFmRate>
```

S1B_IW_SLC__1SDV_20170807T010035_20170807T010102_006827_00C043_B7E0 (IPF `2.82`)

```xml
<azimuthFmRate>
    <azimuthTime>2017-08-07T01:00:35.780368</azimuthTime>
     <t0>5.334643585041154e-03</t0>
    <azimuthFmRatePolynomial count="3">-2.329768045614056e+03 4.502942424994464e+05 -7.864570826763810e+07</azimuthFmRatePolynomial>
</azimuthFmRate>
```

This fix detects if the field `azimuthFmRatePolynomial` exists in the input ET, and try different parsing strategy depending on the existence.

### CODE REVIEW

**Review by gshiroma — APPROVED** [review]
LGTM. Thank you for the update, @seongsujeong . I just have some minor comments:  please avoid these types of qualitative statements, especially in CAPS LOCK: `VERY old`. Also, please follow PEP8 rules for inline comments (two spaces before and a single space after `#`): https://peps.python.org/pep-0008/#inline-comments .

File: src/s1reader/s1_reader.py [hunk 1]
```diff
@@ -56,7 +56,12 @@ def parse_polynomial_element(elem, poly_name):
 
     half_c = 0.5 * isce3.core.speed_of_light
     r0 = half_c * float(elem.find('t0').text)
-    coeffs = [float(x) for x in elem.find(poly_name).text.split()]
+
+    if elem.find(poly_name) is None:  # some data with old IPF version (e.g. 2.36)
```
Comment (vbrancat) [inline src/s1reader/s1_reader.py:60]: We are usually leaving comments at the top of each line of code, to make sure to be in the limits of the pep8 convention.

Comment (vbrancat) [inline src/s1reader/s1_reader.py:60]: I would probably say "with IPF < 2.36"

**Review by vbrancat — APPROVED** [review]
LGTM, just some nit-picking :)

File: src/s1reader/s1_reader.py (hunk 1 above)
Comment (seongsujeong) [reply src/s1reader/s1_reader.py:60]: @vbrancat Thanks for taking a look at the PR.
What I currently know about the relationship between IPF version and the format difference (for AZ FM rate polynomial) is that the change has taken place somewhere >2.36 and <2.82 (by taking a look at the S1 data I've processed so far). I took a look at the three different versions of S1 Product specification, but was not able to find a clear answer about the exact IPF version from which the format has changed.
I've put more context and description to the comment.

Comment (seongsujeong) [reply src/s1reader/s1_reader.py]: Thanks for our suggestion. I've checked the style guide (https://peps.python.org/pep-0008/#inline-comments), and concluded that the inline comments would be helpful to describe which line is for the old format (and which one is for the new format). I've made sure that it passed the codacy.

**Pull request merged**